src/scanner.cc

```diff
@@ -295,6 +336,10 @@ struct Scanner {
     }
   }
 
+  bool is_iden_char(char c) {
+    return NON_ALPHA_NUM_CHARS.find(c) == NON_ALPHA_NUM_CHARS.end();
```

Comment:
I think the simplest and fastest way to do this would be to store the `NON_IDENTIFIER_CHARS` as a simple C array and search it using [`memchr(3)`](http://man7.org/linux/man-pages/man3/memchr.3.html), which is usually written in assembly using vectorized instructions so its crazy fast.

Maybe something like:

```cpp
const char NON_IDENTIFIER_CHARS[] = {
  ' ',
  '\t',
  '\n',
  // ...
};

bool is_iden_char(char c) {
  return memchr(&NON_IDENTIFIER_CHARS, c, sizeof(NON_IDENTIFIER_CHARS)) == NULL;
}
```